clients/StateChannelWallet.ts

```diff
@@ -174,20 +168,17 @@ export class StateChannelWallet {
   /**
    * getBalance checks the blockchain for the current balance of the wallet.
    */
-  async getBalance(): Promise<number> {
-    // TODO: Casting a bigint to a number is dangerous
-    return Number(
-      await this.chainProvider.getBalance(this.scBridgeWalletAddress),
-    );
+  async getBalance(): Promise<bigint> {
+    return await this.chainProvider.getBalance(this.scBridgeWalletAddress);
   }
 
-  async getIntermediaryBalance(): Promise<number> {
-    return Number(this.intermediaryBalance);
+  get intermediaryBalance(): bigint {
```

Comment:
Reminder: this syntax is a bit magic. To consumers, it appears as a straightforward property access. But internally, it calls a method.